rlinf/workers/rollout/server/server_rollout_worker.py

```diff
@@ -0,0 +1,348 @@
+        }
+
+        # Count files and calculate total size
+        if self.storage_dir.exists():
+            for file_path in self.storage_dir.iterdir():
+                if file_path.is_file():
+                    stats["total_files"] += 1
+                    stats["total_size_bytes"] += file_path.stat().st_size
+
+        return stats
+
+class ServerRolloutWorker(Worker):
+    """
+    ServerRolloutWorker that supports both HTTP API and Channel interfaces.
+    It can receive training data from router's feedback_worker via HTTP
+    and also work with OnlineCodingRunner via Channel interface.
+
+    Key features:
+    - Unified data processing for both HTTP and Channel inputs
+    - Automatic rollout processing after server startup
+    - Compatible with OnlineCodingRunner interface
+    """
+
+    def __init__(self, cfg: DictConfig):
+        Worker.__init__(self)
+
+        self._cfg = cfg
+
+        # Initialize tokenizer for text processing
+        self._tokenizer = AutoTokenizer.from_pretrained(self._cfg.rollout.model_dir)
+
+        # Configuration
+        self._server_host = cfg.server.tracking_rollout.get('host', '0.0.0.0')
+        self._server_port = cfg.server.tracking_rollout.get('port', 8082)
+
+        # Unified data source for both HTTP and Channel data
+        # max_queue_size = getattr(self._cfg.server, 'max_queue_size', 1000)
+        max_queue_size = 1000
+        self._data_source = asyncio.Queue(maxsize=max_queue_size)
+
+        # Initialize training data storage
+        # storage_config = getattr(self._cfg, 'storage', None)
+        storage_config = None
+        if storage_config is not None:
+            storage_config = dict(storage_config)
+        self._storage = TrainingDataStorage(storage_config)
+
+        # Processing configuration
+        self._max_new_tokens = getattr(self._cfg.algorithm.sampling_params, 'max_new_tokens', 512)
+        self._batch_size = cfg.data.rollout_batch_size * cfg.algorithm.group_size
+
+        # Processing control
+        self._track_data_enable = False
+
+        # Output channel for continuous processing
+
+        # Setup FastAPI routes
+        self._setup_routes()
+        self._server_task = None
+
+    def _setup_routes(self):
+        """Setup FastAPI routes."""
+        app = FastAPI(title="OnlineRouterWorker", version="1.0.0")
+        app.add_route("/api/training/submit", self._handle_track, methods=["POST"])
+
+        # Init the HTTP server
+        self._server = uvicorn.Server(uvicorn.Config(
+            app,
+            host=self._server_host,
+            port=self._server_port,
+            log_level="info"
+        ))
+
+    def server_start(self):
+        """Start service."""
+        assert self._server_task is None
+
+        # Start server in background task
+        self._server_task = asyncio.create_task(self._server.serve())
+
+        self.log_info(f"service started on {self._server_host}:{self._server_port}")
+
+    def server_stop(self):
```

Comment:
As both `online_router_worker.py` and `server_rollout_worker.py`, how about extracting server as a dedicated class, and pass it to the two workers?